dragonfly/player/player.go

```diff
@@ -404,6 +406,37 @@ func (p *Player) Heal(health float64, source healing.Source) {
 	})
 }
 
+// updateFallState is called to update the entities falling state.
+func (p *Player) updateFallState(distanceThisTick float64) {
+	if p.OnGround() {
+		if p.fallDistance.Load() > 0 {
+			p.fall(p.fallDistance.Load())
+			p.fallDistance.Store(0)
+		}
+	} else if distanceThisTick < p.fallDistance.Load() {
+		p.fallDistance.Store(p.fallDistance.Load() - distanceThisTick)
+	} else {
+		p.fallDistance.Store(0)
+	}
+}
+
+// fall is called when a falling entity hits the ground.
+func (p *Player) fall(fallDistance float64) {
+	fallDamage := fallDistance - 3
+	for _, e := range p.Effects() {
+		if _, ok := e.(effect.JumpBoost); ok {
+			fallDamage -= float64(e.Level())
+		}
+	}
+	fallDamage = math.Ceil(fallDamage)
+
+	if fallDamage < 0.5 {
```

Comment:
It ceils first, then checks if it's smaller than 0.5, but that will never be the case unless it's 0 exactly. You'd probably want to move the ceil below this block.

Reply:
Moved.